utils/utils.go

```diff
@@ -0,0 +1,189 @@
+package utils
+
+// #include "limits.h"
+import "C"
+
+import (
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gluster/glusterd2/errors"
+)
+
+// IsLocalAddress checks whether a given host/IP is local
+func IsLocalAddress(host string) bool {
+	var finalHost string
+	found := false
+
+	addrs, e := net.LookupHost(host)
+	if e != nil {
+		log.Error(e.Error())
+		return found
+	}
+	for _, addr := range addrs {
+		names, err := net.LookupAddr(addr)
+		if err != nil {
+			log.Error(e.Error())
+			return found
+		}
+		for _, name := range names {
+			// LookupHost/LookupAddr appends a '.' at the end of the
+			// hostname, not sure why though, so this is hack as of
+			// now
+			finalHost = name[0 : len(name)-1]
+			if finalHost == host ||
+				finalHost == "localhost.localdomain" ||
+				finalHost == "localhost" {
+				found = true
+				return found
+			}
+		}
+	}
+	return found
+}
+
+// ParseHostAndBrickPath parses the host & brick path out of req.Bricks list
+func ParseHostAndBrickPath(brickPath string) (string, string) {
+	idx := strings.LastIndex(brickPath, ":")
+	if idx == -1 {
+		log.Error("Invalid brick path, it should be in the form of host:path")
+		return "", ""
+	}
+	hostname := brickPath[0:idx]
+	path := brickPath[idx+1 : len(brickPath)]
+
+	return hostname, path
+}
+
+//ValidateBrickPathLength validates the length of the brick path
+func ValidateBrickPathLength(brickPath string) int {
+	//TODO : Check whether PATH_MAX is compatible across all distros
+	if len(filepath.Clean(brickPath)) >= syscall.PathMax {
+		log.Error("brickpath is too long")
+		return -1
+	}
+	return 0
+}
+
+//ValidateBrickSubDirLength validates the length of each sub directories under
+//the brick path
+func ValidateBrickSubDirLength(brickPath string) int {
+	subdirs := strings.Split(brickPath, string(os.PathSeparator))
+	// Iterate over the sub directories and validate that they don't breach
+	//  _POSIX_PATH_MAX validation
+	for _, subdir := range subdirs {
+		if len(subdir) >= C._POSIX_PATH_MAX {
```

Comment:
syscall.PathMax is 4096, here the requirement is POSIX_PATH_MAX which is 256 chars. I haven't come across any constant which represents POSIX'es PATH_MAX.